### PR TITLE
Merge HWGS and other changes

### PR DESCRIPTION
**Pull request by TotallyNotElite**
Changes were tested with @BenCat07 in casual.

### CODE REVIEW

File: src/hacks/Achievement.cpp [hunk 1]
```diff
@@ -212,6 +222,33 @@ bool equip_hats_fn(std::vector<int> hats, std::pair<int, int> classes)
     return true;
 }
 
+static std::vector<int> hat_list = { 302, 940, 941 };
+// TNE didn't want me to put "epic_" infront of it :(
```
Comment (TotallyNotElite) [inline src/hacks/Achievement.cpp]: Yes.

File: src/hacks/Aimbot.cpp [hunk 2]
```diff
@@ -97,6 +98,12 @@ bool IsBacktracking()
     return (aimkey ? aimkey.isKeyDown() : true) && shouldBacktrack();
 }
 
+// Am I holding Hitman's Heatmaker ?
+static bool CarryingHeadmaker()
```
Comment (TotallyNotElite) [inline src/hacks/Aimbot.cpp]: Headmaker is incorrect

File: data/menu/nullifiedcat/antiaim.xml [hunk 3]
```diff
@@ -46,6 +46,7 @@
                     <Option name="FakeOffset" value="15"/>
                     <Option name="FakeEdge" value="16"/>
                     <Option name="FakeHeck" value="17"/>
+                    <Option name="FakeSideways" value="18"/>
```
Comment (BenCat07) [inline data/menu/nullifiedcat/antiaim.xml]: Wait why

File: src/hacks/Aimbot.cpp [hunk 4]
```diff
@@ -21,6 +21,7 @@ static settings::Boolean enable{ "aimbot.enable", "false" };
 static settings::Button aimkey{ "aimbot.aimkey.button", "<null>" };
 static settings::Int aimkey_mode{ "aimbot.aimkey.mode", "1" };
 static settings::Boolean autoshoot{ "aimbot.autoshoot", "1" };
+static settings::Boolean autoreload{ "aimbot.autoshoot.activate-headmaker", "false" };
```
Comment (BenCat07) [inline src/hacks/Aimbot.cpp]: Possibly add to GUI, not high priority though

File: src/hacks/Aimbot.cpp [hunk 5]
```diff
@@ -97,6 +98,12 @@ bool IsBacktracking()
     return (aimkey ? aimkey.isKeyDown() : true) && shouldBacktrack();
 }
 
+// Am I holding Hitman's Heatmaker ?
+static bool CarryingHeadmaker()
+{
+    return CE_INT(LOCAL_W, netvar.iItemDefinitionIndex) == 752;
```
Comment (BenCat07) [inline src/hacks/Aimbot.cpp]: Could have used HasWeapon

File: src/hacks/Spam.cpp [hunk 6]
```diff
@@ -341,17 +341,16 @@ void createMove()
...
+                current_index = UniformRandomInt(0, source->size() - 1);
+                while (current_index == last_index)
                 {
-                    current_index = rand() % source->size();
+                    current_index = UniformRandomInt(0, source->size() - 1);
                 }
             }
             last_index             = current_index;
-            std::string spamString = source->at(current_index);
+            std::string spamString = (*source)[current_index];
```
Comment (BenCat07) [inline src/hacks/Spam.cpp]: There is a reason i used .at

File: src/ipc.cpp [hunk 7]
```diff
@@ -183,16 +183,17 @@ void UpdateTemporaryData()
 
     if (data.connected)
     {
-        IClientEntity *player = g_IEntityList->GetClientEntity(g_IEngine->GetLocalPlayer());
-        if (player)
+        if (CE_GOOD(LOCAL_E))
```
Comment (BenCat07) [inline src/ipc.cpp]: LOCAL_E being good and the client ent not being nullptr are two very different things. Please use the previous method

File: src/hacks/Aimbot.cpp (hunk 2 above)
Comment (BenCat07) [reply src/hacks/Aimbot.cpp]: Yeah

File: src/hacks/Aimbot.cpp (hunk 5 above)
Comment (BenCat07) [reply src/hacks/Aimbot.cpp]: Nevermind.

**Pull request merged**